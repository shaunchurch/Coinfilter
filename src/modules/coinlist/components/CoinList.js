// @flow
import React, { Component } from 'react';
import styled from 'styled-components';
import CoinRow from './CoinRow';
import type { Coin } from '../coinlist.types';

type Props = {
  title: string,
  coins: Array<Coin>,
  handleClick: Function
};

class CoinList extends Component<Props> {
  render() {
    const { coins, handleClick, title } = this.props;

    return (
      <div>
        <Title>
          {title}
        </Title>
        <List>
          {coins.length
            ? coins.map(coin =>
                <CoinRow key={coin.id} coin={coin} handleClick={handleClick} />
              )
            : <li>Loading coins...</li>}
        </List>
      </div>
    );
  }
}

const List = styled.ul`
  max-width: 620px;
  margin: 0 auto;
  padding: 0;
`;

const Title = styled.h1`
  text-align: center;
  font-family: menlo, monaco, courier, monospace;
`;

export default CoinList;
