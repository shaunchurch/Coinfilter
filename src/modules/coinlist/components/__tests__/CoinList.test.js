import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CoinList from '../CoinList';
import CoinJSON from '../../../../testutils/data/Coin.json';

describe('<CoinList />', () => {
  it('should render a loading item when list is empty', () => {
    const wrapper = shallow(<CoinList coins={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render a list of coins when supplied', () => {
    const wrapper = shallow(<CoinList coins={CoinJSON} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
