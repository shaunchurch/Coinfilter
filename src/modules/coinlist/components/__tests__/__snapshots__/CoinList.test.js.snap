// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoinList /> should render a list of coins when supplied 1`] = `
<div>
  <styled.h1 />
  <styled.ul>
    <CoinRow
      coin={
        Object {
          "24h_volume_usd": 1511610000,
          "available_supply": 16526487,
          "id": "bitcoin",
          "last_updated": 1503791971,
          "market_cap_usd": 71929882019,
          "name": "Bitcoin",
          "percent_change_1h": -0.18,
          "percent_change_24h": -0.45,
          "percent_change_7d": 5.29,
          "price_btc": 1,
          "price_usd": 4352.4,
          "rank": 1,
          "symbol": "BTC",
          "total_supply": 16526487,
        }
      }
    />
    <CoinRow
      coin={
        Object {
          "24h_volume_usd": 514181000,
          "available_supply": 94263186,
          "id": "ethereum",
          "last_updated": 1503791974,
          "market_cap_usd": 31472781162,
          "name": "Ethereum",
          "percent_change_1h": 0.26,
          "percent_change_24h": 0.58,
          "percent_change_7d": 13.11,
          "price_btc": 0.0768412,
          "price_usd": 333.882,
          "rank": 2,
          "symbol": "ETH",
          "total_supply": 94263186,
        }
      }
    />
  </styled.ul>
</div>
`;

exports[`<CoinList /> should render a loading item when list is empty 1`] = `
<div>
  <styled.h1 />
  <styled.ul>
    <li>
      Loading coins...
    </li>
  </styled.ul>
</div>
`;
